src/renderer/commands/quickOpen.js

```diff
@@ -0,0 +1,196 @@
+import path from 'path'
+import { ipcRenderer } from 'electron'
+import { isChildOfDirectory } from 'common/filesystem/paths'
+import bus from '../bus'
+import { delay } from '@/util'
+import FileSearcher from '@/node/fileSearcher'
+
+const MD_EXTENSION = /\.(?:markdown|mdown|mkdn|md|mkd|mdwn|mdtxt|mdtext|text|txt)$/i
+const SPECIAL_CHARS = /[\[\]\\^$.\|\?\*\+\(\)\/]{1}/g // eslint-disable-line no-useless-escape
+
+// The quick open command
+class QuickOpenCommand {
+  constructor (rootState) {
+    this.id = 'file.quick-open'
+    this.description = 'File: Quick Open'
+    this.shortcut = null
+
+    this.subcommands = []
+    this.subcommandSelectedIndex = -1
+
+    // Reference to folder and editor and project state.
+    this._editorState = rootState.editor
+    this._folderState = rootState.project
+
+    this._directorySearcher = new FileSearcher()
+    this._cancelFn = null
+  }
+
+  search = async query => {
+    // Show opened files when no query given.
+    if (!query) {
+      return this.subcommands
+    }
+
+    const { _cancelFn } = this
+    if (_cancelFn) {
+      _cancelFn()
+      this._cancelFn = null
+    }
+
+    const timeout = delay(300)
```

Comment:
Why need to delay 300ms?

Reply:
To throttle the function because I don't want to spawn and kill a process after each key down event.